src/domain.py

```diff
@@ -357,3 +361,70 @@ def build_prometheus_config(charm_config):
 
     logger.debug("Build prom config: {}".format(prometheus_config))
     return prometheus_config
+
+
+def _prometheus_http_api_call(
+        model_name, app_name, method, endpoint, return_response=True):
+    host = "{0}.{1}.svc:{2}".format(
+        app_name, model_name, PROMETHEUS_ADVERTISED_PORT
+    )
+    conn = http.client.HTTPConnection(host)
+    conn.request(method=method, url=endpoint)
+    logger.debug("Calling Prom API: {0} {1}".format(method, endpoint))
+
+    # TODO: Handle the un-available API endpoint case
+    response = conn.getresponse()
+    if response.status != 200:
+        logger.error("API returned error: {0}".format(
+            response.read()
+        ))
+        raise PrometheusAPIError("Reload API returned error")
+
+    if return_response:
+        try:
+            return json.loads(response.read())
+        except (ValueError, TypeError) as e:
+            logger.error("Prom API returned non-JSON: {0}".format(e))
+            raise PrometheusAPIError("Non-JSON response returned")
+
+
+def prometheus_config_reload(model_name, app_name):
+    return _prometheus_http_api_call(
+        model_name, app_name, 'POST', '/-/reload', return_response=False
+    )
```

Comment:
Error handling should be done here. If the http call returned a status code of `< 200 or >= 300` then raise `PrometheusAPIError("Reload API returned error")`. Same with attempting to `json.loads()` the response body. These suggested changes will also make your unit tests simpler because you won't have to mock out `json` and `http.client.HTTPConnection` for this fuction.